src/components/TodoList/TodoList.tsx

```diff
@@ -1,100 +1,96 @@
-        </td>
-      </tr>
+      if (filter === 'completed') {
+        return todo.completed;
+      }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">6</td>
-        <td className="is-vcentered" />
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-danger">
-            qui ullam ratione quibusdam voluptatem quia omnis
-          </p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
+      return true;
+    }) || null;
+
+  if (!filteredTodos) {
+    throw new Error('Todos is missing');
+  }
 
-      <tr data-cy="todo" className="">
-        <td className="is-vcentered">8</td>
-        <td className="is-vcentered">
-          <span className="icon" data-cy="iconCompleted">
-            <i className="fas fa-check" />
-          </span>
-        </td>
-        <td className="is-vcentered is-expanded">
-          <p className="has-text-success">quo adipisci enim quam ut ab</p>
-        </td>
-        <td className="has-text-right is-vcentered">
-          <button data-cy="selectButton" className="button" type="button">
+  const searchTodo = filteredTodos.filter(todo =>
+    todo.title.toLowerCase().includes(search.toLowerCase()),
+  );
+
+  return (
+    <table className="table is-narrow is-fullwidth">
+      <thead>
+        <tr>
+          <th>#</th>
+          <th>
             <span className="icon">
-              <i className="far fa-eye" />
+              <i className="fas fa-check" />
             </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+          </th>
+          <th>Title</th>
+          <th> </th>
+        </tr>
+      </thead>
+
+      <tbody>
+        {searchTodo?.map(todo => {
+          return (
+            <tr key={todo.id} data-cy="todo" className="">
+              <td className="is-vcentered">{todo.id}</td>
+              {!todo.completed ? (
+                <td className="is-vcentered" />
+              ) : (
+                <td className="is-vcentered">
+                  <span className="icon" data-cy="iconCompleted">
+                    <i className="fas fa-check" />
+                  </span>
+                </td>
+              )}
+              <td className="is-vcentered is-expanded">
+                <p
+                  className={`${!todo.completed ? 'has-text-danger' : 'has-text-success'}`}
+                >
+                  {todo.title}
+                </p>
+              </td>
+              <td className="has-text-right is-vcentered">
+                <button
+                  data-cy="selectButton"
+                  className="button"
+                  type="button"
+                  onClick={() => onModalButtonClick(todo.id)}
```

Comment:
The naming convention for the event handler function could be improved. According to the conventions, Handler functions should start with 'handle' and the prop that provides the event handler function should start with 'on'. So, 'onModalButtonClick' could be renamed to 'handleModalButtonClick'.

Reply:
but i pass it as a prop